app/models/concerns/orchestration/dhcp.rb

```diff
@@ -94,8 +94,9 @@ def queue_dhcp_update
 
   # do we need to update our dhcp reservations
   def dhcp_update_required?
-    # IP Address / name changed
-    return true if ((old.ip != ip) or (old.hostname != hostname) or (old.mac != mac) or (old.subnet != subnet))
+    # IP Address / name changed, or 'rebuild' action is triggered and DHCP record on the smart proxy is not present/identical.
+    return true if ((old.ip != ip) or (old.hostname != hostname) or (old.mac != mac) or (old.subnet != subnet) or
```

Comment:
that's true, but that would have to be changed in code that is not introduced by this PR, I'd rather keep changes in this area minimal until we extract this to new objects...